### ISSUE TITLE
[Bug] CollectionView doesn't always snap to start

### ISSUE THREAD

**Issue by davidortinau** [xamarin/Xamarin.Forms#8608]
### Description

I have a `CollectionView` with horizontal linear layout that should snap to the start of the item. This works much of the time, but it's very easy to land on an item that is cut and doesn't snap.

### Steps to Reproduce

1. Run the app
2. Go to the Book a Flight
3. Scroll the boxes at the top horizontally until you get an item that doesn't snap

### Basic Information

- Version with issue: 4.4.0-pre2
- Platform Target Frameworks: 
  - Android: 10.0

### Screenshots

![image](https://user-images.githubusercontent.com/41873/69343091-8788a980-0c32-11ea-94d8-72c6f46fd81a.png)

### Reproduction Link

https://github.com/davidortinau/FlyMe/blob/bug/cv_snap/FlyMe/FlyMe/Views/BookingPage.xaml#L30

**Comment by jfversluis** [comment]
Agreed there is something weird going on here. I also have a lot of times where I scroll through the items and the `CarouselView` seems to scroll a few items back. Also land on an item that is only half visible is a possibility.

**Comment by jsuarezruiz** [comment]
@rmarinho Do you think this issue should have the blocker label?

**Comment by AswinPG** [comment]
> Agreed there is something weird going on here. I also have a lot of times where I scroll through the items and the `CarouselView` seems to scroll a few items back. Also land on an item that is only half visible is a possibility.
> 
> On iOS it's a bit different but also doesn't work. Snapping does not seem to work and I can't even reach the last item in the `CarouselView`.
> 
> ![Untitled](https://user-images.githubusercontent.com/939291/69857909-2c3d6380-1291-11ea-8eae-26d67fad81e8.gif)
> 
> This is tested with the latest nightlies.


![20201109-181417](https://user-images.githubusercontent.com/29278621/98543308-4ad91680-22b8-11eb-95af-a8b32f5649c1.gif)

**Comment by AswinPG** [comment]
> > Agreed there is something weird going on here. I also have a lot of times where I scroll through the items and the `CarouselView` seems to scroll a few items back. Also land on an item that is only half visible is a possibility.
> > On iOS it's a bit different but also doesn't work. Snapping does not seem to work and I can't even reach the last item in the `CarouselView`.
> > ![Untitled](https://user-images.githubusercontent.com/939291/69857909-2c3d6380-1291-11ea-8eae-26d67fad81e8.gif)
> > This is tested with the latest nightlies.
> 
> ![20201109-181417](https://user-images.githubusercontent.com/29278621/98543308-4ad91680-22b8-11eb-95af-a8b32f5649c1.gif)

CarouselView Behaves like a collectionview when scrolled slowly

**Comment by AswinPG** [comment]
Also if we use peekarea insets to view more than one item in the carouselview and scroll to more than one one item it scrolls us back to the next item of the current selected item and not the one which we scrolled it to (Using xamarin forms 4.8)
![20201109-182904](https://user-images.githubusercontent.com/29278621/98544453-09e20180-22ba-11eb-9b5b-1b393418a5a0.gif)
